### PR TITLE
Specified validation error when encrypted value is given when secure is false. #2259

### PR DESCRIPTION
**Pull request by varshavaradarajan**

### CODE REVIEW

**Comment by varshavaradarajan** [comment]
#2259

File: config/config-api/src/com/thoughtworks/go/config/EnvironmentVariableConfig.java [hunk 1]
```diff
@@ -204,9 +205,16 @@ public boolean hasName(String variableName) {
 
     public boolean validateTree(ValidationContext validationContext) {
         validate(validationContext);
+        validateEncryptedValue(validationContext);
         return errors().isEmpty();
     }
 
+    public void validateEncryptedValue(ValidationContext validationContext) {
```
Comment (ketan) [inline config/config-api/src/com/thoughtworks/go/config/EnvironmentVariableConfig.java]: Does this need to be public?

**Comment by ketan** [comment]
LGTM, can you trigger the PR build?

**Comment by ketan** [comment]
@varshavaradarajan — I'm merging this in, if you see it being problematic we can pair on fixing it, or roll it back.

**Pull request merged**